zerver/tests/test_users.py

```diff
@@ -1852,6 +1852,32 @@ def test_stream_recipient_info(self) -> None:
         )
         self.assertEqual(info["stream_push_user_ids"], {hamlet.id})
 
+        # Now have Hamlet mute the stream and unmute the topic,
+        # which shouldn't omit him from stream_push_user_ids.
+        sub.is_muted = True
+        sub.save()
+
+        add_topic_visibility_policy(
+            user_profile=hamlet,
+            stream_id=stream.id,
+            recipient_id=recipient.id,
+            topic_name=topic_name,
+            visibility_policy=UserTopic.UNMUTED,
+        )
+
+        info = get_recipient_info(
+            realm_id=realm.id,
+            recipient=recipient,
+            sender_id=hamlet.id,
+            stream_topic=stream_topic,
+        )
+        self.assertEqual(info["stream_push_user_ids"], {hamlet.id})
+
+        # Now unmute the stream and remove topic visibility_policy.
+        sub.is_muted = False
```

Comment:
Not in this PR, but it would probably be nice to split this already 200+ lines long test into multiple smaller tests (one for each scenario).